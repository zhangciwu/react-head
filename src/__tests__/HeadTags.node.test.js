import * as React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import { HeadProvider, HeadTag, Title, Style, Meta, Link } from '../';

describe('HeadTag during server rendering', () => {
  it('renders nothing and adds tags to headTags context array', () => {
    const arr = [];
    const globalCss = `p {
      color: #121212;
    }`;
    const markup = renderToStaticMarkup(
      <HeadProvider headTags={arr}>
        <div>
          Yes render
          <HeadTag tag="test" content="testing tag">
            No render
          </HeadTag>
          <Title>Title</Title>
          <Style>{globalCss}</Style>
          <Link href="index.css" />
          <Meta charset="utf-8" />
        </div>
      </HeadProvider>
    );
    expect(markup).toMatchSnapshot();
    expect(arr).toMatchSnapshot();
  });

  it('render only the last title', () => {
    const arr = [];
    renderToStaticMarkup(
      <HeadProvider headTags={arr}>
        <div>
          <Title>Title 1</Title>
        </div>
        <div>
          <Title>Title 2</Title>
        </div>
        <div>
          <Title>Title 3</Title>
        </div>
      </HeadProvider>
    );
    expect(arr).toMatchSnapshot();
  });
});
