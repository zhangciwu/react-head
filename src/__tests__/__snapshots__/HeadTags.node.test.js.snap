// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadTag during server rendering render only the last title 1`] = `
Array [
  <title
    data-rh=""
  >
    Title 3
  </title>,
]
`;

exports[`HeadTag during server rendering renders nothing and adds tags to headTags context array 1`] = `"<div>Yes render</div>"`;

exports[`HeadTag during server rendering renders nothing and adds tags to headTags context array 2`] = `
Array [
  <test
    content="testing tag"
    data-rh=""
  >
    No render
  </test>,
  <title
    data-rh=""
  >
    Title
  </title>,
  <style
    data-rh=""
  >
    p {
      color: #121212;
    }
  </style>,
  <link
    data-rh=""
    href="index.css"
  />,
  <meta
    charset="utf-8"
    data-rh=""
  />,
]
`;
